eras/alonzo/impl/src/Cardano/Ledger/Alonzo/Rules/Utxow.hs

```diff
@@ -312,24 +303,24 @@ requiredSignersAreWitnessed txbody witsKeyHashes = do
 {-  scriptIntegrityHash txb = hashScriptIntegrity pp (languages txw) (txrdmrs txw)  -}
 ppViewHashesMatch ::
   forall era.
-  ( ValidateScript era,
+  ( AlonzoEraTxBody era,
     ExtendedUTxO era,
-    Core.Script era ~ Script era,
-    Core.Tx era ~ ValidatedTx era,
-    HasField "scriptIntegrityHash" (Core.TxBody era) (StrictMaybe (ScriptIntegrityHash (Crypto era))),
-    HasField "_costmdls" (Core.PParams era) CostModels
+    Script era ~ AlonzoScript era,
+    Tx era ~ AlonzoTx era,
+    HasField "_costmdls" (PParams era) CostModels
   ) =>
-  Core.Tx era ->
-  Core.TxBody era ->
-  Core.PParams era ->
+  Tx era ->
+  TxBody era ->
+  PParams era ->
   UTxO era ->
   Set (ScriptHash (Crypto era)) ->
   Test (UtxowPredicateFail era)
-ppViewHashesMatch tx txbody pp utxo sNeeded = do
+ppViewHashesMatch tx txBody pp utxo sNeeded = do
+  -- FIXME: No need to supply txBody as a separate argument: txBody = tx ^. bodyTxL
```

Comment:
good catch! :facepalm: 

I think we usually use `TODO` in these circumstances, but `FIXME` certainly makes sense. This is probably something else we could be more consistent about...